packages/store/src/StoreCore.sol

```diff
@@ -64,49 +64,67 @@ library StoreCore {
     }
   }
 
+  /**
+   * Get the schema for the given tableId
+   */
+  function getValueSchema(bytes32 tableId) internal view returns (Schema valueSchema) {
+    valueSchema = Schema.wrap(Tables.getValueSchema(tableId));
+    if (valueSchema.isEmpty()) {
+      revert IStoreErrors.StoreCore_TableNotFound(tableId, string(abi.encodePacked(tableId)));
+    }
+  }
+
   /**
    * Check if a table with the given tableId exists
    */
   function hasTable(bytes32 tableId) internal view returns (bool) {
-    return Tables.getValueSchema(tableId) != bytes32(0);
+    return Tables.getFieldLayout(tableId) != bytes32(0);
   }
 
   /**
-   * Register a new table with key schema, value schema, key names, and field names
+   * Register a new table with key field layout, value field layout, key names, and value names
    */
   function registerTable(
     bytes32 tableId,
+    FieldLayout fieldLayout,
     Schema keySchema,
     Schema valueSchema,
     string[] memory keyNames,
-    string[] memory fieldNames
+    string[] memory valueNames
```

Comment:
for context, we went with "field names" instead of "value names" because we didn't have a good solution for the name of values ("field values" since "value values" sounds weird)